llama_stack/cli/stack/_build.py

```diff
@@ -266,8 +266,8 @@ def run_stack_build_command(args: argparse.Namespace) -> None:
     if args.run:
         config_dict = yaml.safe_load(run_config.read_text())
         config = parse_and_maybe_upgrade_config(config_dict)
-        if not os.path.exists(config.external_providers_dir):
-            os.makedirs(config.external_providers_dir, exist_ok=True)
+        if not os.path.exists(str(config.external_providers_dir)):
```

Comment:
```suggestion
        if not config.external_providers_dir.exists():
```
We shouldn't need to call `os.path` in the first place since `external_providers_dir` is a Path.